File: frontend/android/SmallSocialAndroidApp/app/src/main/java/de/aljoshavieth/smallsocialandroidapp/Post.java
package de.aljoshavieth.smallsocialandroidapp;

import java.util.ArrayList;

public class Post {
    private String id;
    private String content;
    private String color;
    private ArrayList<Comment> comments;
    private User author;
    private long time;

    public Post(String id, String content, String color, ArrayList<Comment> comments, User author, long time) {
        this.id = id;
        this.content = content;
        this.color = color;
        this.comments = comments;
        this.author = author;
        this.time = time;
    }

    public String getId() {
        return id;
    }

    public String getContent() {
        return content;
    }

    public String getColor() {
        return color;
    }

    public ArrayList<Comment> getComments() {
        return comments;
    }

    public User getAuthor() {
        return author;
    }

    public long getTime() {
        return time;
    }

    public void setComments(ArrayList<Comment> comments) {
        this.comments = comments;
    }

}

File: backend/content-service/src/main/kotlin/de/aljoshavieth/userservice/models/Post.kt
package de.aljoshavieth.userservice.models

import kotlinx.serialization.Serializable

@Serializable
data class Post(val id: String, val content: String,val color:String, val comments: Array<Comment>, val author: User, val time: Long) : ContentServiceModel{
    // auto generated...
    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as Post

        if (id != other.id) return false
        if (content != other.content) return false
        if (content != other.color) return false
        if (!comments.contentEquals(other.comments)) return false
        if (author != other.author) return false
        if(time != other.time) return false

        return true
    }

    override fun hashCode(): Int {
        var result = id.hashCode()
        result = 31 * result + content.hashCode()
        result = 31 * result + color.hashCode()
        result = 31 * result + comments.contentHashCode()
        result = 31 * result + author.hashCode()
        result = 31 * result + time.hashCode()
        return result
    }


}
